### PR TITLE
Use repo assignable / mentionable users in finder

### LINKED ISSUE #423: Filter assignees to current repo
## Issue Description

Type: *feature request*

### Describe what happened (or what feature you want)

When assigning a user to a PR or Issue it searches for all users on Github. Is it possible to only show users with access to your repo? Like in the Github Web UI you only get relevant users in the dropdown and not everyone on Github.
Hopefully it would speed it up a little bit as well :)

### PR DESCRIPTION
**Pull request by wd60622**
### Describe what this PR does / why we need it

Use repo's assignable / mentionable users to populate finder instead of general GitHub search.

TODO:

- [x] Add a configuration option so that the previous behavior can be used if desired
- [ ] Implement for the other pickers / refactor user list logic to separate file. Thoughts on this @pwntester?
- [ ] Update any documentation

### Does this pull request fix one issue?

Closes #423 

### Describe how you did it

Pull out the finder creation logic

### Describe how to verify it

Use `Octo assignee add` or `Octo reviewer add` to see the new behavior.

### CODE REVIEW

File: lua/octo/pickers/telescope/provider.lua [hunk 1]
```diff
@@ -767,76 +767,110 @@ end
...
-              else
-                vim.list_extend(orgs[user.login].teams, user.teams.nodes)
-              end
-            end
-          end
-        end
+  -- local finder = finders.new_dynamic {
+  --     entry_maker = entry_maker.gen_from_user(),
+  --     fn = get_user_requester(),
+  -- }
```
Comment (wd60622) [inline lua/octo/pickers/telescope/provider.lua:864]: TODO: Toggle with configuration option? What are your thoughts here @pwntester

File: lua/octo/pickers/telescope/provider.lua [hunk 2]
```diff
@@ -767,76 +767,110 @@ end
 --
 -- ASSIGNEES
 --
+local function get_user_requester()
```
Comment (wd60622) [inline lua/octo/pickers/telescope/provider.lua:770]: This function seem to be duplicated for the pickers and the next function would have to be duplicated as well. Is there a file location that would serve to house this logic?

File: lua/octo/pickers/telescope/provider.lua (hunk 1 above)
Comment (wd60622) [reply lua/octo/pickers/telescope/provider.lua]: went will a key "users" which takes values "search", "mentionable", or "assignable"
Let me know your thoughts on this approach

File: lua/octo/pickers/telescope/provider.lua (hunk 2 above)
Comment (pwntester) [reply lua/octo/pickers/telescope/provider.lua:772]: Good question, we could create a `pickers/shared` for this kind of code

**Review by pwntester — APPROVED** [review]
It would be great to move the shared code to a shared location so other pickers can reuse it. But since there is no other implementation for fzf-lua yet, Im ok if you want to merge this and let the implementator for fzf-lua to refactor that code.

**Comment by pwntester** [comment]
@milogert just a heads up about a new implementation of the picker for choosing users in case you can port it to fzf-lua

**Comment by wd60622** [comment]
Might be good to add to the readme before merge if you think this config change is reasonable @pwntester

**Comment by pwntester** [comment]
Yep, please update the docs so users know the three options and the differences between them

**Comment by wd60622** [comment]
I've added to the README. Let me know what you think about the location @pwntester

**Comment by pwntester** [comment]
> I've added to the README. Let me know what you think about the location @pwntester

That should be enough, thanks!

**Comment by GrimalDev** [comment]
Tested the integration on my config. You guys rule! It is a banger and the documentation is crystal clear.

**Comment by pwntester** [comment]
Thanks @wd60622!

**Comment by milogert** [comment]
> @milogert just a heads up about a new implementation of the picker for choosing users in case you can port it to fzf-lua

Thanks! I can start taking a look at this 👌

**Pull request merged**